pkg/kubectl/cmd/cmd.go

```diff
@@ -252,6 +255,12 @@ var (
 
 // NewKubectlCommand creates the `kubectl` command and its nested children.
 func NewKubectlCommand(f cmdutil.Factory, in io.Reader, out, err io.Writer) *cobra.Command {
+	if f := os.Getenv("KUBECTL_FEATURE_GATES"); f != "" {
```

Comment:
Thanks for asking this, it made me realize that I hadn't actually Submitted my comment explaining my reasoning and it was just hanging out in a text box in another tab waiting for me to close my browser. Here it is: https://github.com/kubernetes/kubernetes/issues/45922#issuecomment-302895410

There's a bit of a catch-22 with setting feature gates on kubectl itself. If we set them with a flag then the gates won't be available prior to flag parsing and so can't gate flags or the command tree.

We definitely should document this, but first let's reach consensus on how the feature gates will be implemented.